enumerable.rb

```diff
@@ -0,0 +1,279 @@
+        accumulator = yield(accumulator, self[i])
+
+        i += 1
+      end
+    else
+      p 'No block given!'
+    end
+    accumulator
+  end
+end
+# rubocop:enable all
+def multiply_els(arr)
+  arr.my_inject(1) { |i, j| i * j }
+end
+
+print '-------------------'
+puts "\n"
+puts 'Start tests'
+puts "\n"
+print '-------odin inject test---------'
+puts "\n"
+p multiply_els([2, 4, 5])
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-----Start enumerable tests------'
+puts "\n"
+print '-------my_each---------'
+puts "\n"
+[5, 3, 7, 300, 800].my_each { |x| p x }
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_each_with_index---------'
+puts "\n"
+[5, 3, 7, 300, 800].my_each_with_index { |x, y| p "#{x} => #{y}" }
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_select---------'
+puts "\n"
+p [1, 2, 3, 4, 5].my_select(&:even?)
+# rubocop:disable all
+p %i[foo bar].my_select { |x| x == :foo }
+# rubocop:enable all
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_all?---------'
+puts "\n"
+# rubocop:disable all
+p %w[ant bear cat].my_all? { |word| word.length >= 3 }
+p %w[ant bear cat].my_all? { |word| word.length >= 4 }
+# rubocop:enable all
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_any?---------'
+puts "\n"
+# rubocop:disable all
+p %w[ant bear cat].my_any? { |word| word.length >= 3 }
+p %w[ant bear cat].my_any? { |word| word.length >= 4 }
+# rubocop:enable all
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_none?---------'
+puts "\n"
+# rubocop:disable all
+p %w[ant bear cat].my_none? { |word| word.length == 5 }
+p %w[ant bear cat].my_none? { |word| word.length >= 4 }
+# rubocop:enable all
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_count---------'
+puts "\n"
+ary = [1, 2, 4, 2]
+p ary.count
+p ary.count(2)
+p ary.count(&:even?)
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_map methods---------'
+puts "\n"
+print '-------my_map with block---------'
+puts "\n"
+# rubocop:disable all
+p [1, 2, 3, 4].my_map { |i| i * i }
+p [1, 2, 3, 4].my_map { 'cat' }
+# rubocop:enable all
+puts "\n"
+print '-------------------'
+print '-------my_map with proc---------'
+puts "\n"
+my_proc = proc { |i| i * i }
+my_proce = proc { 'cat' }
+p [1, 2, 3, 4].my_map_1(&my_proc)
+p [1, 2, 3, 4].my_map_1(&my_proce)
+puts "\n"
+print '-------------------'
+print '-------my_map with proc or block---------'
+puts "\n"
+bloc = proc { 'cat' }
+# rubocop:disable all
+p [1, 2, 3, 4].my_map_2 { |i| i * i }
+# rubocop:enable all
+p [1, 2, 3, 4].my_map_2(&bloc)
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------------------'
+puts "\n"
+print '-------my_inject---------'
+puts "\n"
+# rubocop:disable all
+p [5, 6, 7, 8, 9, 10].my_inject { |sum, n| sum + n }
+# rubocop:enable all
+p [5, 6, 7, 8, 9, 10].my_inject(1) { |product, n| product * n }
+puts "\n"
+print '-------------------'
```

Comment:
Please remove this, if you want you can add to a separate file and add it to git ignore